docs/infra/treatment-service.md

```diff
@@ -0,0 +1,132 @@
+
+You can also refer to [values.yaml](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service/values.yaml)
+for a minimal configuration that needs to be provided for XP Treatment Service installation.
+
+```shell
+$ helm install xp-treatment xp/xp-treatment \
+    --values=path/to/helm/chart/values/file.yaml
+```
+
+Notice that you can specify the *Treatment Service configuration* [values](#configuration) under the `xpTreatment` 
+field in the Helm chart values file like below: 
+
+```yaml
+xpTreatment:
+  config:
+    Port: 8080
+    ManagementService:
+      URL: https://caraml-dev.io/api/xp/v1
+      AuthorizationEnabled: true
+    ...
+```
+
+These configuration values would be saved in a `.yaml` file within a 
+[secret](https://kubernetes.io/docs/concepts/configuration/secret/) that gets mounted automatically onto the 
+Treatment Service pod, where it will be read by the Treatment Service.
+
+##### 2.1 Configure the Treatment Service to use a Google Cloud Provider (GCP) Service Account (Optional) 
+
+In particular, should you need to use a GCP service account for authenticating, one recommended way to set this up 
+is to utilise [volumes](https://kubernetes.io/docs/concepts/storage/volumes/) to mount the GCP service account key file, 
+which itself should be saved within a [secret](https://kubernetes.io/docs/concepts/configuration/secret/) in the 
+Kubernetes cluster, into the Treatment Service [pod](https://kubernetes.io/docs/concepts/workloads/pods/). Adding the 
+environment variable `GOOGLE_APPLICATION_CREDENTIALS` with the path to the mounted key file would then allow the 
+Treatment Service to access it.
+
+When deploying the Treatment Service with Helm, however, it is convenient to specify `extraVolumes`, 
+`extraVolumeMounts` and `extraEnvs` in the Helm chart values (see
+[values.yaml](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service/values.yaml#L59)), as they 
+automatically get added to the Treatment 
+Service deployment's volumes, volume mounts and environment variables respectively, via templates in the Helm chart 
+(see 
+[deployment.yaml](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service/templates/deployment.yaml#L37)).
+
+*Note that this is NOT the only one way to set the GCP service account key file to allow the Treatment Service to
+access it. There are a wide variety of other methods, such as containerising the Treatment Service binary together with 
+the GCP service account key file, but we will not be listing all of them here.*
+
+##### 3. Update the Helm Release
+
+To subsequently update a deployed release of `xp-treatment`, you may run something like the following, which updates 
+not only the `xp-treatment`'s image but also its config values:
+
+```shell
+$ helm upgrade xp-treatment xp/xp-treatment \
+    --set xpTreatment.image.tag=${NEW_VERSION_TAG} \
+    --values=path/to/updated/helm/chart/values/file.yaml
+```
+
+## Treatment Service Plugin 
```

Comment:
Can we also have a new section `## Monitoring the Treatment Service` after all the existing sections? Key points that come to mind:
* There are regular Kubernetes metrics to monitor the deployments that the users can leverage. (The chart itself isn't tied to Istio or any other framework, so don't need to mention those.)
* Summarize the custom metrics produced by the Treatment service (and how the prefix will defer for the 2 modes of deployment), something like what we have here: https://github.com/caraml-dev/turing/blob/main/docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md

Reply:
Just added a new section for this :)